url/url.h

```diff
@@ -5,14 +5,21 @@
 #ifndef URL_URL_H_
 #define URL_URL_H_
 
+#include "util/base_parser.h"
+#include "util/string.h"
+
 #include <array>
 #include <cstdint>
+#include <map>
 #include <optional>
 #include <string>
+#include <string_view>
 #include <variant>
 
 namespace url {
 
+void icu_init();
```

Comment:
This is called internally where it's needed (I think?). Do we need it in the public header as well?

Reply:
This isn't a reference to an ICU function, this is a function that sets the ICU data directory and tests that the data was loaded correctly. In the future, it'll also pull a data dir from whatever config system we end up going with. It belongs in something like an i18n lib, but i just stuck it here for now since this is the only thing using ICU.

Reply:
On second thought, maybe we should just call it once from `//engine` and be done with it? I'm not sure where (or when, for that matter) I should be doing initialization for something like this, which only needs to happen once for the entire engine.

Reply:
I meant that `UrlParser::domain_to_ascii` already calls `icu_init`, so when would a user need to call it themselves? At least right now it doesn't seem like it would be needed outside of url_parser.cpp?

Reply:
I can hide it until we need it elsewhere then.